pkg/cmd/service/service.go

```diff
@@ -21,68 +19,191 @@ import (
-	GRPCTLSCertPath string `envconfig:"OPENFGA_GRPC_TLS_CERT_PATH"`
-	GRPCTLSKeyPath  string `envconfig:"OPENFGA_GRPC_TLS_KEY_PATH"`
+type Config struct {
+	// If you change any of these settings, please update the documentation at https://github.com/openfga/openfga.dev/blob/main/docs/content/intro/setup-openfga.mdx
 
-	HTTPEnabled     bool   `default:"true" envconfig:"OPENFGA_HTTP_ENABLED"`
-	HTTPTLSEnabled  bool   `default:"false" envconfig:"OPENFGA_HTTP_TLS_ENABLED"`
-	HTTPTLSCertPath string `envconfig:"OPENFGA_HTTP_TLS_CERT_PATH"`
-	HTTPTLSKeyPath  string `envconfig:"OPENFGA_HTTP_TLS_KEY_PATH"`
+	OpenFGA    OpenFGAConfig
+	Database   DatabaseConfig
+	GRPC       GRPCConfig
+	HTTP       HTTPConfig
+	Authn      AuthnConfig
+	Log        LogConfig
+	Playground PlaygroundConfig
+}
 
-	// Authentication. Possible options: none,preshared,oidc
-	AuthMethod string `default:"none" split_words:"true"`
+// DefaultConfig returns the OpenFGA server default configurations.
+func DefaultConfig() Config {
+	return Config{
+		OpenFGA: OpenFGAConfig{
+			MaxTuplesPerWrite:             100,
+			MaxTypesPerAuthorizationModel: 100,
+			ChangelogHorizonOffset:        0,
+			ResolveNodeLimit:              25,
+		},
+		Database: DatabaseConfig{
+			Engine:       "memory",
+			MaxCacheSize: 100000,
+		},
+		GRPC: GRPCConfig{
+			Enabled: true,
+			Addr:    ":8081",
+			TLS:     TLSConfig{Enabled: false},
+		},
+		HTTP: HTTPConfig{
+			Enabled:                true,
+			Addr:                   ":8080",
+			TLS:                    TLSConfig{Enabled: false},
+			UpstreamRequestTimeout: 3 * time.Second,
+			CORSAllowedOrigins:     []string{"*"},
+			CORSAllowedHeaders:     []string{"*"},
+		},
+		Authn: AuthnConfig{
+			Method: "none",
+		},
+		Log: LogConfig{
+			Format: "text",
+		},
+		Playground: PlaygroundConfig{
+			Enabled: false,
+			Port:    3000,
+		},
+	}
+}
 
-	// Shared key authentication
-	AuthPresharedKeys []string `default:"" split_words:"true"`
+// GetServiceConfig returns the OpenFGA server configuration based on the values provided in the server's 'config.yaml' file.
+// The 'config.yaml' file is loaded from '/etc/openfga', '$HOME/.openfga', or the current working directory. If no configuration
+// file is present, the default values are returned.
+func GetServiceConfig() (Config, error) {
 
-	// OIDC authentication
-	AuthOIDCIssuer   string `default:"" split_words:"true"`
-	AuthOIDCAudience string `default:"" split_words:"true"`
+	config := DefaultConfig()
 
-	// Logging. Possible options: text,json
-	LogFormat string `default:"text" split_words:"true"`
+	viper.SetConfigName("config")
+	viper.SetConfigType("yaml")
 
-	PlaygroundEnabled bool `default:"true" split_words:"true"`
-	PlaygroundPort    int  `default:"3000" split_words:"true"`
+	configPaths := []string{"/etc/openfga", "$HOME/.openfga", "."}
+	for _, path := range configPaths {
+		viper.AddConfigPath(path)
+	}
+	viper.AutomaticEnv()
```

Comment:
Do we still need this?

Reply:
I guess if you want to provide env variables and they automatically map to the config values then it's ok. In a subsequent PR we will add support for other options (CLI flags and possibly env variables with a better mapping strategy)